lib/examples/component-with-ng-component-outlet.spec.ts

```diff
@@ -0,0 +1,53 @@
+import { Component, Injectable, NgModule } from '@angular/core';
+import { Shallow } from '../shallow';
+
+////// Module Setup //////
+@Component({
+  template: '<h1>Step</h1>',
+  selector: 'step'
+})
+class StepComponent {}
+
+@Injectable()
+class StepService {
+  getSteps() {
+    return [StepComponent];
+  }
+}
+
+@Component({
+  selector: 'step-display',
+  template: `
+    <div *ngFor="let step of stepService.getSteps()">
+      <ng-template [ngComponentOutlet]="step"></ng-template>
+    </div>
+`,
+})
+class StepDisplayComponent {
+  constructor(public stepService: StepService) { }
+}
+
+@NgModule({
+  declarations: [StepDisplayComponent, StepComponent],
+  providers: [StepService],
+  entryComponents: [StepComponent]
+})
+class StepModule {}
+//////////////////////////
+
+describe('component with ngComponentOutlet and entry component', () => {
+  let shallow: Shallow<StepDisplayComponent>;
+
+  beforeEach(() => {
+    shallow = new Shallow(StepDisplayComponent, StepModule)
+      .dontMock(StepService)
+      .dontMock(StepComponent);
```

Comment:
It kinda bugs me that the only way this works is if we `dontMock` the `StepComponent`.

I'm going to see if I can figure out a way to allow a mock before I merge this.